OpenRA.Game/Game.cs

```diff
@@ -594,55 +597,57 @@ static void InnerLogicTick(OrderManager orderManager)
 				Cursor.Tick();
 			}
 
-			var worldTimestep = world == null ? Timestep : world.IsLoadingGameSave ? 1 : world.Timestep;
+			var worldTimestep = orderManager.SuggestedTimestep;
+
 			var worldTickDelta = tick - orderManager.LastTickTime;
-			if (worldTimestep != 0 && worldTickDelta >= worldTimestep)
+			if (worldTimestep == 0 || worldTickDelta < worldTimestep)
+				return;
+
+			using (new PerfSample("tick_time"))
 			{
-				using (new PerfSample("tick_time"))
+				if (world == null || !orderManager.ShouldUseCatchUp || !orderManager.LobbyInfo.GlobalSettings.UseNewNetcode)
```

Comment:
The `world == null` and `!orderManager.ShouldUseCatchUp` (which appears to be equivalent to `OrderManager.Connection != NetworkConnection`) cases look like they will be a problem here. What is our route to removing these once we want to remove the legacy netcode and tick logic in the future?